hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/DynamicConfigRollingUpgradeTest.java

```diff
@@ -73,4 +74,26 @@ public void testThrowsExceptionWhenAddingCountDownLatchConfigInClusterVersion39(
         hazelcastInstance.getConfig().addCountDownLatchConfig(new CountDownLatchConfig(randomMapName()));
     }
 
+    @Test(expected = UnsupportedOperationException.class)
+    public void testThrowsExceptionWhenAddingFlakeIdConfigInClusterVersion39() {
+        // system properties are cleared automatically by the Hazelcast Runner
+        System.setProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION, Versions.V3_9.toString());
+        HazelcastInstance hazelcastInstance = createHazelcastInstance();
+
+        hazelcastInstance.getConfig().addFlakeIdGeneratorConfig(new FlakeIdGeneratorConfig(randomName()));
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testThrowsExceptionWhenAddingPNCounterConfigInClusterVersion39() {
+        // system properties are cleared automatically by the Hazelcast Runner
+        System.setProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION, Versions.V3_9.toString());
+        HazelcastInstance hazelcastInstance = createHazelcastInstance();
+
+        hazelcastInstance.getConfig().addPNCounterConfig(new PNCounterConfig(randomName()));
+    }
+
+    @Override
+    protected HazelcastInstance createHazelcastInstance() {
```

Comment:
Leftover? Or should this be:
```java
    private HazelcastInstance createHazelcastInstance(Version version) {
        // System Properties are cleared automatically by the Hazelcast Runner
        System.setProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION, version.toString());
        return createHazelcastInstance();
    }
```
so we can simplify the tests like:
```java
HazelcastInstance hazelcastInstance = createHazelcastInstance(Versions.V3_8);
HazelcastInstance hazelcastInstance = createHazelcastInstance(Versions.V3_9);
```

Reply:
Good catch! Fixed this and other comments.